### PR TITLE
Add changelog for 0.15.0

### PR DESCRIPTION
**Pull request by ellismg**
Work in progress.

### CODE REVIEW

**Comment by ellismg** [comment]
I'll be flushing out the rest of the packages tonight.  @hausdorff Please let me know what you want to say for Kubernetes.The full set of changes from the previous release is here: https://github.com/pulumi/pulumi-kubernetes/compare/v0.14.0...v0.15.0-rc4

**Comment by ellismg** [comment]
I assume we want to regenerate the API documentation as part of this change, but I don't know how to do that.

File: install/changelog.md [hunk 1]
```diff
@@ -52,6 +57,86 @@ title: "Change Log"
...
+
+In addition to the 0.15.0 CLI release, we've released new versions of all of our packages.
+
+### Pulumi CLI
+
+This release brings many large improvements to the Pulumi CLI. Major features include:
+
+#### Parallelism
+
+Pulumi now preforms resource operations in parallel, when possible. By default, Pulumi allows ten parallel operations, but this can be controlled with the `-p` flag to `pulumi`. Presently, deletes are still processed sequentaly, but this will be changed in a future release. It is possible that your program may have implicit dependencies between two resources and the change to preform operations in parallel by default will cause issues. If this happens, we recommend that you set dependsOn on property in `resourceOptions` parameter to any resource. You may also pass `-p=1` to disable parallelism. Parallelism is only supported for nodejs programs today.
```
Comment (lukehoban) [inline install/changelog.md]: Does `-p=1` disable parallelism, or is that `=0`?  @swgillespie?

Comment (lukehoban) [reply install/changelog.md]: Preforms => performs.

File: install/changelog.md [hunk 2]
```diff
@@ -52,6 +57,86 @@ title: "Change Log"
...
+
+#### First class providers
+
+Pulumi now allows resource providers to be created and configured as part of your program. In addition to the ambiently available provider for each resource, you can also create an explicit version of the provider and configure it as you wish. This is helpful, for example, if you want to create some resources in a different region from your main deployment. We have an [example](https://github.com/pulumi/pulumi-aws/blob/master/examples/multiple-regions/index.ts) that shows how to use this feature and we are excited to see what you build with it.
```
Comment (lukehoban) [inline install/changelog.md]: Perhaps mention also “or configure the inputs to one provider (e.g. Kubernetes) using an output of a resource created by another (e.g.  Azure Kubernetes Service)”.

File: install/changelog.md [hunk 3]
```diff
@@ -52,6 +57,86 @@ title: "Change Log"
...
+1. Remove the `main` and `typings` directives from `package.json`, as well as the `build` script.
+2. Remove the `bin` folder that contained your previously compiled code.
+3. You may remove the dependency on `typescript` from your `package.json` as well.
+
+To use this new support, you must upgrade your `@pulumi/pulumi` version to 0.15.0 as well. While a `tsconfig.json` file is no longer required other tools like VS Code behave better when it is present, so you'll probably want to keep it.
+
+#### Miscellaneous improvements
+
+- The CLI no longer emits warnings if it can't detect metadata about your git enlistement (for example, what GitHub project it coresponds to).
+- The CLI now only warns about adding a plaintext configuration in specific cases instead of for every value you add.
```
Comment (lukehoban) [inline install/changelog.md]: In cases where it appears likely you may be storing a secret.

**Comment by lukehoban** [comment]
> I assume we want to regenerate the API documentation as part of this change, but I don't know how to do that.

Tracking that piece in https://github.com/pulumi/docs/pull/525.

File: install/changelog.md [hunk 4]
```diff
@@ -52,6 +57,86 @@ title: "Change Log"
 
 > See [known issues](../reference/known-issues.html) for currently known issues and workarounds.
 
+## v0.15.0 {#v150}
+
+Released on August 13, 2018
+
+In addition to the 0.15.0 CLI release, we've released new versions of all of our packages.
+
+### Pulumi CLI
```
Comment (joeduffy) [inline install/changelog.md]: We should also include major features in the service.

At the very least, this should include our GitHub App. /cc @chrsmith as he agreed to write this (I suggest we all just push to this PR).

File: install/changelog.md [hunk 5]
```diff
@@ -52,6 +57,86 @@ title: "Change Log"
...
+#### Miscellaneous improvements
+
+- The CLI no longer emits warnings if it can't detect metadata about your git enlistement (for example, what GitHub project it coresponds to).
+- The CLI now only warns about adding a plaintext configuration in specific cases instead of for every value you add.
+
+### @pulumi/pulumi 0.15.0
+
+#### Closure capturing improvements
+
+We've improved our closure capturing logic, which should allow you to write more idomatic code in lambda functions that are uploaded to the cloud. Previously, if you wanted to use a module, we required you to write either `require('module')` or `await import('module')` inside your lambda function. In addition, if you wanted to use a helper you defined in another file, you had to require that module in your function as well. With these changes, the following code now works:
```
Comment (justinvp) [inline install/changelog.md]: idomatic => idiomatic

File: install/changelog.md (hunk 1 above)
Comment (joeduffy) [inline install/changelog.md]: "this will be changed in a future release" => link to GitHub issue

**Comment by joeduffy** [comment]
I just pushed some edits.

File: install/changelog.md [hunk 6]
```diff
@@ -52,6 +57,100 @@ title: "Change Log"
...
+
+Released on August 13, 2018
+
+In addition to the 0.15.0 CLI release, we've released new versions of all of our packages.
+
+### Pulumi CLI
+
+This release brings many large improvements to the Pulumi CLI. Major features include:
+
+#### Kubernetes
```
Comment (joeduffy) [inline install/changelog.md]: @hausdorff Can you look this over and/or suggest edits?

File: install/changelog.md [hunk 7]
```diff
@@ -52,6 +57,100 @@ title: "Change Log"
...
+* Support for provisioning and deploying to clusters in a single program
+* Deploying Helm Charts using the `helm.v2.Chart` type
+
+To use Pulumi with Kubernetes, please [visit our Getting Started page on Kubernetes](/install/kubernetes.html).
+
+#### Parallelism
+
+Pulumi now performs resource creates and updates in parallel, driven by dependencies in the resource graph. (Parallel deletes are coming in a future release.) If your program has implicit dependencies that Pulumi does not already see as dependencies, it's possible parallel will cause ordering issues. If this happens, you may set the `dependsOn` on property in the `resourceOptions` parameter to any resource. By default, Pulumi allows 10 parallel operations, but the `-p` flag can be used to override this. `-p=1` disables parallelism altogether. Parallelism is supported for Node.js and Go programs, and Python support will come in a future release.
+
+#### First class providers
```
Comment (joeduffy) [inline install/changelog.md]: Nit, inconsistent casing (all others are Upper Cased)

File: install/changelog.md (hunk 6 above)
Comment (joeduffy) [reply install/changelog.md]: This obviously doesn't belong in the CLI section ... @lukehoban How do we elevate this to a top-level thing? I don't want to bury the lede on our most exciting part of this release!

**Comment by lukehoban** [comment]
I pushed some additions for libraries.  @hausdorff I didn't yet add anything for Kubernetes, would be great if you could take a first pass.

File: install/changelog.md (hunk 6 above)
Comment (lukehoban) [reply install/changelog.md]: Once we've have all the content here - I'll reorganize a bit.  We can pull a few of the "big" things up to the top.

**Comment by chrsmith** [comment]
I'm working on the docs for our GitHub app now, hopefully to land just in time to add a quick blurb to the release notes.

File: install/changelog.md (hunk 1 above)
Comment (swgillespie) [reply install/changelog.md]: Anything <= 1 disables parallelism.

File: install/changelog.md [hunk 8]
```diff
@@ -52,6 +57,153 @@ title: "Change Log"
...
+    const statusText = (await axios.default.get("https://pulumi.io")).statusText;
+    res.write(`GET https://pulumi.io/ == ${statusText}`).end();
+});
+```
+
+#### Default value for configuration package
+
+The `pulumi.Config` object can now be created without an argument. When no argument is supplied, the value of the current project is used. This means that application level code can simply do `new pulumi.Confg()` without passing any argument. For library authors, you should continue to pass the name of your package as an argument.
+
+### @pulumi/kubernetes 0.15.0
```
Comment (lukehoban) [inline install/changelog.md]: Should we delete this section now that we've got Kubernetes in the intro?

**Pull request merged**